projects/user_publication/citation.py

```diff
@@ -40,33 +41,35 @@ def update_scopus_citation(pub, dry_run=True):
         search_results = []
         if search and search.results:
             for x in search.results:
-                if (
-                    PublicationUtils.how_similar(x.title.lower(), pub.title.lower())
-                    >= PublicationUtils.SIMILARITY
-                ):
+                if (PublicationUtils.are_similar(x.title.lower(), pub.title.lower())):
                     search_results.append(x)
         if (len(search_results)) > 0:
             scopus_pub = search_results[0]
     if scopus_pub:
-        existing_scopus_cites = pub.scopus_citations
+        # Returns a tuple of (object, created)
+        existing_scopus_source = pub.source.get_or_create(name=Publication.SCOPUS)[0]
+        logger.info((
+            f"update scopus citation number for "
+            f"{pub.title} (id: {pub.id}) "
+            f"from {existing_scopus_source.citation_count} "
+            f"to {scopus_pub.citedby_count}"
+        ))
         if not dry_run:
-            pub.scopus_citations = scopus_pub.citedby_count
-            pub.save()
-        logger.info(
-            (
-                f"update scopus citation number for "
-                f"{pub.title} (id: {pub.id}) "
-                f"from {existing_scopus_cites} "
-                f"to {scopus_pub.citedby_count}"
-            )
-        )
+            existing_scopus_source.citation_count = scopus_pub.citedby_count
+            existing_scopus_source.save()
```

Comment:
Database updates should be done inside `transaction.atomic()`. We don't do it in a lot of places in portal... but we should!